### PR TITLE
Add preview of old project versions

### PR DESCRIPTION
**Pull request by joshlory**

### CODE REVIEW

**Comment by davidsbailey** [comment]
Hi Josh, looking now. Is there a pivotal item? Does this expose previews in the UI, and if so can you include a screenshot?

**Comment by davidsbailey** [comment]
What is the plan for remix, will clicking remix while we are looking at an old version eventually cause the old version to be remixed?

File: apps/src/templates/VersionHistory.jsx [hunk 1]
```diff
@@ -48,11 +48,19 @@ var VersionHistory = React.createClass({
...
+    location.reload();
+  },
+
+  /**
    * Called when the user chooses a previous version to restore.
    * @param versionId
    */
   onChooseVersion: function (versionId) {
-    location.search += '&version=' + versionId;
+    // TODO: Use Dave's client api when it's finished.
```
Comment (davidsbailey) [inline apps/src/templates/VersionHistory.jsx:62]: Please remove this TODO, unless you can make it more specifically point to work we're actually planning to do

**Comment by davidsbailey** [comment]
Nice feature Josh! Additional comments after finishing a first pass:
1. The Remix button should probably be disabled when viewing an old version, unless we plan to make it remix whatever version you're currently viewing in which case there should be a pivotal item to track that
2. This feature could really use a test. I think a regular UI test seems best but an eyes test could also work.

**Comment by joshlory** [comment]
Clicking remix causes the old version to be remixed.  I'm not sure this is a bad thing, but I can remove it if you think it's confusing.  I could see there being some value in being able to make a remix from an old version, even if you don't want to change the current version.

**Comment by joshlory** [comment]
I updated the existing Version History UI test to check for the presence of the version-preview button.  It's not as easy to test the button itself since the preview opens in a new tab.

**Comment by davidsbailey** [comment]
Remixing the old version sounds great! :+1:  Without names of old versions to guide me, this seems way less scary than actually restoring any of the old versions.

It looks like it is possible to test tab behavior in selenium: http://stackoverflow.com/questions/12729265/switch-tabs-using-selenium-webdriver-with-java 

If you don't want to deal with switching tabs (which seems reasonable), then it should still be possible to test the contents of the page at the URL which would open in a new tab by navigating to it directly.

**Comment by davidsbailey** [comment]
Also the tab which shows the older version probably shouldn't show the Version History button, especially because if I click it and then click Restore Version nothing happens.

**Comment by davidsbailey** [comment]
CORRECTED: Per offline discussion, OK to merge after:
1. hiding Version History button when viewing the older version
2. opening a pivotal item to add UI tests for the new view-only-at-version mode, provided manual testing has been done to ensure that view does not affect app state

**Comment by davidsbailey** [comment]
LGTM

**Pull request merged**